### PR TITLE
SC-4260 - sentry-nuxt

### PR DESCRIPTION
**Pull request by ABialoskorski**
# Description
Add Sample Rate to sentry in nuxt

## Links to Tickets or other pull requests

- https://ticketsystem.schul-cloud.org/browse/SC-4260
## Changes

Added sample rate to env in nuxt config

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

**Comment by adrianjost** [comment]
@ABialoskorski 
image

**Comment by ABialoskorski** [comment]
> @ABialoskorski
> image

done :)

File: nuxt.config.js [hunk 1]
```diff
@@ -24,6 +24,8 @@ module.exports = {
 			process.env.JWT_SHOW_TIMEOUT_WARNING_SECONDS,
 		JWT_TIMEOUT_SECONDS: process.env.JWT_TIMEOUT_SECONDS,
 		SC_THEME: process.env.SC_THEME,
+		SENTRY_SAMPLE_RATE: parseFloat(process.env.SENTRY_SAMPLE_RATE) || 1.0,
```
Comment (tofixx) [inline nuxt.config.js:27]: parsefloat does not work for undefined (if the value is not defined). 
```suggestion
		SENTRY_SAMPLE_RATE: process.env.SENTRY_SAMPLE_RATE ?parseFloat(process.env.SENTRY_SAMPLE_RATE) : 1.0,
```

File: src/plugins/sentry.js [hunk 2]
```diff
@@ -12,6 +13,14 @@ export default ({ app, store }) => {
 			scope.setTag({ schoolId: user.schoolId });
 		}
 	});
+	if (env.SENTRY_DSN) {
```
Comment (tofixx) [inline src/plugins/sentry.js:16]: I'm not sure if this is already anywhere else already defined...

File: nuxt.config.js (hunk 1 above)
Comment (ABialoskorski) [reply nuxt.config.js:27]: nice ;) done

File: src/plugins/sentry.js [hunk 3]
```diff
@@ -1,4 +1,6 @@
 import Theme from "@theme/config";
+const env = require("../../nuxt.config");
+const { Configuration } = require("@schul-cloud/commons");
```
[the remark this answers is not in the document]
Comment (ABialoskorski) [reply src/plugins/sentry.js:3]: ok, done ;)

File: src/plugins/sentry.js [hunk 4]
```diff
@@ -12,6 +14,14 @@ export default ({ app, store }) => {
 			scope.setTag({ schoolId: user.schoolId });
 		}
 	});
+	if (Configuration.has("SENTRY_DSN")) {
+		Sentry.init({
+			dsn: Configuration.get("SENTRY_DSN"),
+			sampleRate: env.SENTRY_SAMPLE_RATE,
```
[the remark this answers is not in the document]
Comment (ABialoskorski) [reply src/plugins/sentry.js:20]: ok, done ;)

**Comment by BlackGhostHD** [comment]
@tofixx ping

File: nuxt.config.js [hunk 5]
```diff
@@ -24,6 +24,9 @@ module.exports = {
 			process.env.JWT_SHOW_TIMEOUT_WARNING_SECONDS,
 		JWT_TIMEOUT_SECONDS: process.env.JWT_TIMEOUT_SECONDS,
 		SC_THEME: process.env.SC_THEME,
+		SENTRY_SAMPLE_RATE: process.env.SENTRY_SAMPLE_RATE
+			? parseFloat(process.env.SENTRY_SAMPLE_RATE)
+			: 1.0,
```
[the remark this answers is not in the document]
Comment (ABialoskorski) [reply nuxt.config.js:29]: "The sampling factor to apply to events. A value of 0.00 will deny sending any events, and a value of 1.00 will send 100% of events." from docs

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -77,6 +77,9 @@ jobs:
         with:
           name: report
           path: integration-tests/reports
+      - name: Unset git config
+          run: |
+            git config --global --unset url."https://${CI_USER_TOKEN}@github.com/".insteadOf
```
[the remark this answers is not in the document]
Comment (ABialoskorski) [reply .github/workflows/test.yml:82]: I've copied that from my other PR, It should be removed.

File: deploy/compose-client_thr.dummy [hunk 7]
```diff
@@ -17,7 +17,6 @@ services:
       - TZ=Europe/Berlin
       - LEGACY_CLIENT_URL=http://client:3100
       - API_URL=https://api.$SYSTEM.schul-cloud.org
-      - SENTRY_DSN=https://18a7cbf1a7ef427ebcfc38429d18ba40@sentry.schul-cloud.dev/7
```
[the remark this answers is not in the document]
Comment (ABialoskorski) [reply deploy/compose-client_thr.dummy:20]: I think so, @tofixx should know the best answet for that as he suggested to delete those secrets

File: docs/0-GettingStarted/1-Build.md [hunk 8]
```diff
@@ -37,9 +37,9 @@ Output Directory: `./dist/nuxt`
 | PROXY_LOG_LEVEL | `silent` | Loglevel of the legacy proxy. Allowed values: `debug`, `info`, `warn`, `error`, `silent` |
 | FEATURE_TEAMS_ENABLED | `false` | Enables Teams feature in sidebar |
 | FEATURE_EXTENSIONS_ENABLED | `false` | Enables Add-Ons in sidebar. Just for N21! |
-| SENTRY_DSN | `false` | If set, errors are reported to sentry. |
```
[the remark this answers is not in the document]
Comment (ABialoskorski) [reply docs/0-GettingStarted/1-Build.md:40]: I think that after deleting all SENTRY_DSN stuff that is not needed

**Comment by ABialoskorski** [comment]
@tofixx bitte Review :)

File: src/plugins/sentry.js [hunk 9]
```diff
@@ -6,12 +6,15 @@ export default ({ app, store }) => {
 		scope.setLevel("warn");
 		scope.setTag("theme", Theme.name);
 		scope.setTag("domain", window.location.host);
+		scope.setTag("sampleRate", process.env.SENTRY_SAMPLE_RATE);
```
Comment (tofixx) [inline src/plugins/sentry.js:9]: here, just reuse the value from nuxt.config.js

Comment (ABialoskorski) [reply src/plugins/sentry.js:9]: done ;)

File: nuxt.config.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (CordlessWool) [reply nuxt.config.js:41]: yes, but our sentry crashes wenn we send 100% ;), so we deside to go to 0.1 by default. I think this should be also the default value in the other reposotories

File: src/plugins/sentry.js [hunk 10]
```diff
@@ -1,17 +1,21 @@
 import Theme from "@theme/config";
+const nuxtConfig = require("../../nuxt.config");
 
 export default ({ app, store }) => {
 	const { user } = store.state.auth;
 	app.$sentry.configureScope((scope) => {
 		scope.setLevel("warn");
 		scope.setTag("theme", Theme.name);
 		scope.setTag("domain", window.location.host);
+		scope.setTag("sampleRate", nuxtConfig.env.SENTRY_SAMPLE_RATE);
```
Comment (tofixx) [inline src/plugins/sentry.js:10]: `sampleRate` does not seem to be a tag for me https://docs.sentry.io/clients/javascript/config/#optional-settings

**Comment by tofixx** [comment]
replaced by https://github.com/hpi-schul-cloud/nuxt-client/pull/1277

**Pull request closed without merging**